MosaicResidentInformationApi.Tests/V1/Gateways/MosaicGatewayTests.cs

```diff
@@ -179,11 +179,61 @@ public void GetAllResidents_ReturnsResidentsUPRN()
                 .Should().Be(address.Uprn.ToString());
         }
 
+        [Test]
+        public void GetAllResidentsWithFirstNameQueryParameter_ReturnsMatchingResident()
+        {
+            var databaseEntity = AddPersonRecordToDatabase(firstname: "ciasom");
+            var databaseEntity1 = AddPersonRecordToDatabase(firstname: "shape");
+            var databaseEntity2 = AddPersonRecordToDatabase(firstname: "Ciasom");
+
+            var listOfPersons = _classUnderTest.GetAllResidents(firstname: "ciasom");
+            listOfPersons.Count.Should().Be(2);
+            listOfPersons.Should().ContainEquivalentOf(databaseEntity.ToDomain());
+            listOfPersons.Should().ContainEquivalentOf(databaseEntity2.ToDomain());
+        }
+
+        [Test]
+        public void GetAllResidentsWithLastNameQueryParameter_ReturnsMatchingResident()
+        {
+            var databaseEntity = AddPersonRecordToDatabase(lastname: "tessellate");
+            var databaseEntity1 = AddPersonRecordToDatabase(lastname: "square");
+            var databaseEntity2 = AddPersonRecordToDatabase(lastname: "Tessellate");
+
+            var listOfPersons = _classUnderTest.GetAllResidents(lastname: "tessellate");
+            listOfPersons.Count.Should().Be(2);
+            listOfPersons.Should().ContainEquivalentOf(databaseEntity.ToDomain());
+            listOfPersons.Should().ContainEquivalentOf(databaseEntity2.ToDomain());
+        }
+
+        [Test]
+        public void GetAllResidentsWithPostCodeQueryParameter_ReturnsMatchingResident()
+        {
+            var databaseEntity = AddPersonRecordToDatabase();
+            var databaseEntity1 = AddPersonRecordToDatabase();
+
+            var address = TestHelper.CreateDatabaseAddressForPersonId(databaseEntity.Id, "E8 1DY");
+            MosaicContext.Addresses.Add(address);
+            MosaicContext.SaveChanges();
+
+            var address1 = TestHelper.CreateDatabaseAddressForPersonId(databaseEntity1.Id, "E81DY");
```

Comment:
I would expect both of these postcodes to be returned, might be best to use a case that is a completely different postcode.

Reply:
This just highlights the type of query we are currently using on the data. If we should be returning both instances in the test the where clause needs to be altered. I don't know what the real data looks like, but the where part of the EF query will need to be updated to avoid missing postcodes due to spacing.